htdocs/core/class/html.form.class.php

```diff
@@ -201,9 +201,10 @@ public function editfieldkey($text, $htmlname, $preselected, $object, $perm, $ty
 	 * @param	string	$formatfunc		Call a specific function to output field in view mode (For example: 'dol_print_email')
 	 * @param	string	$paramid		Key of parameter for id ('id', 'socid')
 	 * @param	string	$gm				'auto' or 'tzuser' or 'tzserver' (when $typeofdata is a date)
+	 * @param	string	$editaction		[=''] use GETPOST default action or set action to edit mode
 	 * @return  string					HTML edit field
 	 */
-	public function editfieldval($text, $htmlname, $value, $object, $perm, $typeofdata = 'string', $editvalue = '', $extObject = null, $custommsg = null, $moreparam = '', $notabletag = 0, $formatfunc = '', $paramid = 'id', $gm = 'auto')
+	public function editfieldval($text, $htmlname, $value, $object, $perm, $typeofdata = 'string', $editvalue = '', $extObject = null, $custommsg = null, $moreparam = '', $notabletag = 0, $formatfunc = '', $paramid = 'id', $gm = 'auto', $editaction = '')
```

Comment:
Why do you need to be able to force the editaction ?

Reply:
@eldy 
When an error occurs the action value is reset and is different of GET / POST action value.
For example in "product lot" card, you can edit "sell by" field and when you click on "Modify" button the GET / POST action is "setsellby" and if an error occurs we want to edit again "sell by" field and show the error message. So the action value is "editsellby" and the GET / POST action value is "setsellby".